### DISCUSSION TITLE
Popup Reminders

### DISCUSSION THREAD

**Discussion by Stuntflyer** [mikeberger/borg_calendar#181] [General]
It would be great if the Popup reminders would open in front of any already opened windows. Can this be done now? If not, would you consider adding the feature sometime in the future? 

Mike

**Comment by mikeberger** [comment]
The popup reminders have always called the API to request to be brought to the front of all windows. This call is just a request to the window manager of the OS. Years ago, the reminders would pop to the front and grab the focus. Whatever you were doing in other apps would be interrupted. Fullscreen apps, like games, might even minimize to show a reminder. It wasn't so great.

I never changed this logic, but both Windows and Linux changed to prevent this. Now, windows might blink the taskbar or do nothing at all, depending on the version. I think most linux window managers also prevent windows from grabbing focus. If you are using linux, look for "Focus Prevention Level" settings. For windows, there may be a secret registry entry - but I'd avoid it.

There are some suggested coding tricks that sound iffy to me.

Are you using windows? How about getting taskbar notifications? I could look into that.

**Comment by Stuntflyer** [comment]
Hi,
I'm using windows 10 21H2. If you could look into taskbar notifications, that would be great. Right now the Borg icon in the taskbar does not blink when there is a notification. 

Mike

**Comment by mikeberger** [comment]
I coded something up and it seems to work fine on windows 11. I don't expect another release for some time, so I uploaded the latest version here: https://sourceforge.net/projects/borg-calendar/files/drops/BorgCalendar-1.10.04.msi/download in case you want to try it.

**Comment by Stuntflyer** [comment]
With this new release, do I need to check an option to have the window in front?

**Comment by mikeberger** [comment]
oops. forgot to mention. there is a new option under the Popup reminders options to enable taskbar reminders. it is disabled by default. if enabled, then you should get windows taskbar notifications for each reminder that hasn't been hidden in the reminder list. You get these whenever the window would try to go to the front. 

I found no acceptable way to force the OS to allow a window to take focus and show itself.

**Comment by Stuntflyer** [comment]
Thank you for working on this. I love this calendar program and it has replaced my Thunderbird calendar which had issues. 

I use the windows image backup to test software. If I then decide not to use the software, I like to restore via the image rather than remove from "Programs and Features" in control panel. It's cleaner that way. This would often mess up the TB calendar where it would be missing events. Even if I restore a calendar backup taken at the time of the image backup. There is something going on between the sqlite file in AppData and my ics backup which I believe is the cause.

Being able to store your db on my data drive was a game changer for me.

Thanks again,

Mike

**Comment by Stuntflyer** [comment]
Btw, The new release works. .

**Comment by mikeberger** [comment]
I'm glad you find the program useful. Make sure you back up your database, and not just the database file. The full export to XML/ZIP is recommended. The occasional export to ICS format is also a good idea.

**Comment by Stuntflyer** [comment]
Will do, thanks!
